Mobl_ios_Callisto-master/MealTrackingPilot/LogCommonMeal/Views/OccasionPicker.swift

```diff
@@ -597,90 +482,88 @@ open class PickerView: UIView {
-
+extension OccasionPicker: UICollectionViewDataSource {
+    
     // MARK: UICollectionViewDataSource
-
+    
     public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
         return numberOfItemsByDataSource * infinityItemsMultiplier
     }
-
+    
     public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         let indexOfSelectedItem = visibleIndexOfSelectedItem()
-
-        let pickerViewCell = collectionView.dequeueReusableCell(withReuseIdentifier: pickerViewCellIdentifier, for: indexPath) as! SimplePickerCollectionViewCell
-
-        let view = delegate?.pickerView?(self, viewForItem: indexPath.item, index: indexForItem(indexPath.item), highlighted: indexPath.item == indexOfSelectedItem, reusingView: pickerViewCell.customView)
-
-        pickerViewCell.backgroundColor = pickerCellBackgroundColor ?? UIColor.clear
-
+        
+        let occasionPickerCell = collectionView.dequeueReusableCell(withReuseIdentifier: occasionPickerCellIdentifier, for: indexPath) as! OccasionPickerCollectionViewCell
+        
+        let view = delegate?.occasionPicker?(self, viewForItem: indexPath.item, index: indexForItem(indexPath.item), highlighted: indexPath.item == indexOfSelectedItem, reusingView: occasionPickerCell.customView)
+        
+        occasionPickerCell.backgroundColor = pickerCellBackgroundColor ?? UIColor.clear
+        
         if let customView = view {
-            pickerViewCell.customView = customView
+            occasionPickerCell.customView = customView
         } else {
             let size = CGSize(span: itemSpan, lateralSpan: itemLateralSpan, direction: scrollingDirection)
-            pickerViewCell.titleLabel.frame = CGRect(origin: .zero, size: size)
-
-            pickerViewCell.contentView.addSubview(pickerViewCell.titleLabel)
-            pickerViewCell.titleLabel.backgroundColor = UIColor.clear
-            pickerViewCell.titleLabel.text = dataSource?.pickerView(self, titleForItem: indexPath.item, index: indexForItem(indexPath.item))
-
-            delegate?.pickerView?(self, styleForLabel: pickerViewCell.titleLabel, highlighted: indexPath.item == indexOfSelectedItem)
+            occasionPickerCell.titleLabel.frame = CGRect(origin: .zero, size: size)
+            
+            occasionPickerCell.contentView.addSubview(occasionPickerCell.titleLabel)
+            occasionPickerCell.titleLabel.backgroundColor = UIColor.clear
+            occasionPickerCell.titleLabel.text = dataSource?.occasionPicker(self, titleForItem: indexPath.item, index: indexForItem(indexPath.item))
+            
+            delegate?.occasionPicker?(self, styleForLabel: occasionPickerCell.titleLabel, highlighted: indexPath.item == indexOfSelectedItem)
         }
-
-        return pickerViewCell
+        
+        return occasionPickerCell
     }
 }
 
 fileprivate extension CGPoint {
```

Comment:
Create a separate for CGPoint extension line CGPoint+Extensions

Reply:
Done